src/message_element_metadata_owner.cc

```diff
@@ -118,26 +118,20 @@ namespace grpc_labview {
         if (metadata->clusterSize != 0)
         {
             return;
-        }    
+        }
         int clusterOffset = 0;
         for (auto element: metadata->_elements)
         {
             auto elementType = element->type;
             auto nestedElement = element;
-            while (elementType == LVMessageMetadataType::MessageValue)
-            {
-                auto nestedMetadata = FindMetadata(element->embeddedMessageName);
-                auto nestedElement = nestedMetadata->_elements.front();
```

Comment:
I think we need to keep the while loop.  What this code is doing is computing cluster element memory offset (the padding of the elements to keep byte alignment).  When there is a nested message the alignment is based on the first element of the message that itself is not a message.

I think the bug is that we are always calling FindMetadata passing in element->embeddedMessageName and not the changing element to be nestedElement for the next iteration of the while loop.

I would chnange is so that we create a currentElement initialized to element and use that for FindMetadata and after each iteration we change currentElement to equal nestedElement.

Reply:
Would `currentElement` then replace usages of `element` following the loop? Otherwise, what is the loop mutating?

Reply:
Also, aren't we already computing cluster offsets for nested messages in lines 135-151 (old), 129-145 (new). Notice the call is recursive